DataFormats/SoATemplate/test/SoABlocks_t.cc

```diff
@@ -0,0 +1,151 @@
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+#define CATCH_CONFIG_MAIN
+#include <catch.hpp>
+
+#include "DataFormats/SoATemplate/interface/SoABlocks.h"
+
+// This file tests the main properties of SoABlocks
+
+GENERATE_SOA_LAYOUT(SoAPositionTemplate,
+                    SOA_COLUMN(float, x),
+                    SOA_COLUMN(float, y),
+                    SOA_COLUMN(float, z),
+                    SOA_SCALAR(int, detectorType))
+
+GENERATE_SOA_LAYOUT(SoAPCATemplate,
+                    SOA_COLUMN(float, eigenvector_1),
+                    SOA_COLUMN(float, eigenvector_2),
+                    SOA_COLUMN(float, eigenvector_3),
+                    SOA_EIGEN_COLUMN(Eigen::Vector3d, candidateDirection))
+
+GENERATE_SOA_LAYOUT(SoATemplate, SOA_SCALAR(int, id), SOA_SCALAR(int, type), SOA_SCALAR(float, energy))
+
+GENERATE_SOA_BLOCKS(SoABlocksTemplate,
+                    SOA_BLOCK(position, SoAPositionTemplate),
+                    SOA_BLOCK(pca, SoAPCATemplate),
+                    SOA_BLOCK(scalars, SoATemplate))
+
+using SoABlocks = SoABlocksTemplate<>;
+// TODO: Implement template version of View and ConstView (ViewTemplate and ConstViewTemplate)
```

Comment:
What do you mean ?

Reply:
Ah, you mean to implement a version that does not use the default template arguments?
OK.
Can you do in this PR, so it's not left incomplete ?

Reply:
It can also be left for a separate PR, there are already other TODO items anyway.